plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcTableHandle.java

```diff
@@ -59,6 +61,7 @@ public final class JdbcTableHandle
 
     private final int nextSyntheticColumnId;
     private final Optional<String> authorization;
+    private final List<JdbcAssignmentItem> updateAssignments;
```

Comment:
If you are adding this here then you need to check in all the places where `JdbcTableHandle` is used that there are no assignments.  Maybe we need another class for table handle to express UPDATE?

Reply:
Yes I am checking this in all places where we create `JdbcTableHandle` from another `JdbcTableHandle` and transfer assignments if presents.

Reply:
Maybe didn't get it for example for sortOrder we check this argument only in delete:
`checkArgument(handle.getSortOrder().isEmpty(), "Unable to delete when sort order is set: %s", handle);`

Reply:
See places like:
<img width="1033" alt="image" src="https://github.com/trinodb/trino/assets/7569403/ba5a934a-39ec-4099-890f-b27577094b90">

There are other places like that where check that no limit should be used or other fields from jdbc handle. For sure you should not have any updateAssignments when doing delete statement. Please check the codebase and update all such places.

Reply:
Checks added